macros/src/track.rs

```diff
@@ -36,7 +36,7 @@ pub fn expand(item: &syn::Item) -> Result<TokenStream> {
             }
 
             let name = &item.ident;
-            let ty = parse_quote! { dyn #name + '__comemo_dynamic };
+            let ty = parse_quote! { dyn #name + Send + Sync + '__comemo_dynamic };
```

Comment:
Perhaps we should generate an impl for both `dyn` and `dyn + Send + Sync` for convenience.